fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java

```diff
@@ -1047,77 +1066,24 @@ private BigDecimal periodicInterestRate(final PaymentPeriodsInOneYearCalculator
-        BigDecimal loanTermFrequencyBigDecimal;
-        if (isForPMT) {
-            loanTermFrequencyBigDecimal = BigDecimal.valueOf(this.repaymentEvery);
-        } else {
-            loanTermFrequencyBigDecimal = calculateLoanTermFrequency(periodStartDate, periodEndDate);
-        }
-        switch (this.interestCalculationPeriodMethod) {
-            case INVALID:
-            break;
-            case DAILY:
-                BigDecimal numberOfDaysInPeriod = BigDecimal.valueOf(ChronoUnit.DAYS.between(periodStartDate, periodEndDate));
-                final BigDecimal oneDayOfYearInterestRate = this.interestRatePerPeriod.divide(loanTermPeriodsInYearBigDecimal, mc)
-                        .divide(divisor, mc);
-                switch (this.repaymentPeriodFrequencyType) {
-                    case INVALID:
-                    break;
-                    case DAYS:
-                        periodicInterestRate = oneDayOfYearInterestRate.multiply(numberOfDaysInPeriod, mc);
-                    break;
-                    case WEEKS:
-                        periodicInterestRate = oneDayOfYearInterestRate.multiply(numberOfDaysInPeriod, mc);
-                    break;
-                    case MONTHS:
-                        if (daysInMonthType.isDaysInMonth_30()) {
-                            numberOfDaysInPeriod = loanTermFrequencyBigDecimal.multiply(BigDecimal.valueOf(30), mc);
-                        }
-                        periodicInterestRate = oneDayOfYearInterestRate.multiply(numberOfDaysInPeriod, mc);
-                    break;
-                    case YEARS:
-                        switch (daysInYearType) {
-                            case DAYS_360:
-                                numberOfDaysInPeriod = loanTermFrequencyBigDecimal.multiply(BigDecimal.valueOf(360), mc);
-                            break;
-                            case DAYS_364:
-                                numberOfDaysInPeriod = loanTermFrequencyBigDecimal.multiply(BigDecimal.valueOf(364), mc);
-                            break;
-                            case DAYS_365:
-                                numberOfDaysInPeriod = loanTermFrequencyBigDecimal.multiply(BigDecimal.valueOf(365), mc);
-                            break;
-                            default:
-                            break;
-                        }
-                        periodicInterestRate = oneDayOfYearInterestRate.multiply(numberOfDaysInPeriod, mc);
-                    break;
-                    case WHOLE_TERM:
-                        log.error("TODO Implement periodicInterestRate for WHOLE_TERM");
-                    break;
-                }
-            break;
-            case SAME_AS_REPAYMENT_PERIOD:
-                periodicInterestRate = this.interestRatePerPeriod.divide(loanTermPeriodsInYearBigDecimal, mc).divide(divisor, mc)
-                        .multiply(loanTermFrequencyBigDecimal);
-            break;
-        }
```

Comment:
This is using interestRatePerPeriod instead of annualNominalInterest which be changed